pmd-apex/src/main/java/net/sourceforge/pmd/lang/apex/rule/performance/AbstractAvoidNodeInLoopsRule.java

```diff
@@ -0,0 +1,62 @@
+/**
+ * BSD-style license; for more info see http://pmd.sourceforge.net/license.html
+ */
+
+package net.sourceforge.pmd.lang.apex.rule.performance;
+
+import net.sourceforge.pmd.lang.apex.ast.ASTBlockStatement;
+import net.sourceforge.pmd.lang.apex.ast.ASTDoLoopStatement;
+import net.sourceforge.pmd.lang.apex.ast.ASTForEachStatement;
+import net.sourceforge.pmd.lang.apex.ast.ASTForLoopStatement;
+import net.sourceforge.pmd.lang.apex.ast.ASTReturnStatement;
+import net.sourceforge.pmd.lang.apex.ast.ASTWhileLoopStatement;
+import net.sourceforge.pmd.lang.apex.rule.AbstractApexRule;
+import net.sourceforge.pmd.lang.ast.AbstractNode;
+import net.sourceforge.pmd.lang.ast.Node;
+
+/**
+ * Base class for any rules that detect operations contained within a loop that could be more efficiently executed by
+ * refactoring the code into a batched execution.
+ */
+abstract class AbstractAvoidNodeInLoopsRule extends AbstractApexRule {
```

Comment:
Should the common constructor code be moved here or left in the concrete classes? I know that sometimes people like that type of configuration in the concrete class.

Reply:
I'd move common code logic to the base class, but leave rule properties at the concrete classes. Right now, I only see rule properties, so keep it as is.
Note: all these code-climate rule properties will be removed with PMD 7 anyway, so don't worry too much about them.